src/suggestions.ts

```diff
@@ -110,11 +111,47 @@ export async function postPatchSuggestion(
       ...lineArgs,
       commit_id: pull_request.head.sha,
       pull_number: pull_request.number,
-      body: `To fix violation '${ps.violationSpec}', Sorald suggests the following:
-${ps.suggestion}
-`,
+      body: await generateSuggestionMessage(ps),
       path: ps.file.toString(),
       start_side: 'RIGHT'
     });
   }
 }
+
+/**
+ * Generate a full, human-friendly suggestion message based on a patch suggestion.
+ *
+ * @param ps - A patch suggestion
+ * @returns Fulfills to a suggestion message
+ */
+export async function generateSuggestionMessage(
+  ps: PatchSuggestion
+): Promise<string> {
+  const [ruleKey] = ps.violationSpec.split(':');
+  const sonarVersion = '6.9.0.23563';
+  const sonarRuleMetadataUrl = `https://raw.githubusercontent.com/SonarSource/sonar-java/${sonarVersion}/java-checks/src/main/resources/org/sonar/l10n/java/rules/java/S${ruleKey}_java.json`;
+  const ruleInfoPage = `https://rules.sonarsource.com/java/RSPEC-${ruleKey}`;
+
+  const ruleMetadata = await httpGetJson(sonarRuleMetadataUrl);
+  return `This code change violates SonarSource rule [${ruleKey}: ${ruleMetadata.title}](${ruleInfoPage}). Sorald suggests the following fix:
+
+${ps.suggestion}
+
+See [Sorald's documentation for details on the repair](https://github.com/SpoonLabs/sorald/blob/master/docs/HANDLED_RULES.md).
+
+Violation specifier: ${ps.violationSpec}
+`;
+}
+
+async function httpGetJson(url: string): Promise<RuleMetadata> {
+  const response: RuleMetadata = await got(url, {
+    responseType: 'json'
+  }).json();
+  return response;
+}
+
+interface RuleMetadata {
+  title: string;
+  type: string;
+  sqKey: string;
```

Comment:
You can remove `type` and `sqKey` as you are not checking if they exist during compile time.

Reply:
Actually, I can just remove the entire interface, it's pretty pointless.

Reply:
Yes, you can but it will throw linting error [when you access `ruleMetadata.title`](https://github.com/SpoonLabs/sorald-buildbreaker/pull/88/files#diff-ee6ba2e4d87823d56031fd978dabc20ae06a5ea01a85aec533a0c191785f1ecaR133).

Reply:
Of course I must specify the type, otherwise it won't compile, but using an interface is redundant.

Note that TypeScript _requires_ types to be consistently used for it to compile, it's not a linting thing. The linter I use here just to detect problems that aren't related to type (e.g. unused variables and imports, and stuff like that).

Reply:
> requires types to be consistently used for it to compile

I think TypeScript compiles even if the compiler finds error related to types. It just notifies the client about the compilation errors.